pydm/data_plugins/pva_plugins/p4p_plugin_component.py

```diff
@@ -0,0 +1,83 @@
+import logging
+from collections import OrderedDict
+
+from p4p.client.thread import Context, Disconnected
+from qtpy.QtCore import Slot
+
+from pydm.utilities.channel import parse_channel_config
+
+from pydm.data_store import DataKeys
+from pydm.data_plugins.plugin import PyDMPlugin, PyDMConnection
+
+from .pva_helper import pre_process, nt_introspection
+
+logger = logging.getLogger(__name__)
+
+
+class PVAContext(object):
+    """
+    Singleton class responsible for holding the pva context.
+    """
+    __instance = None
+
+    def __init__(self):
+        if self.__initialized:
+            return
+        self.__initialized = True
+        self.context = Context('pva', maxsize=2, unwrap=False)
+
+    def __new__(cls, *args, **kwargs):
+        if cls.__instance is None:
+            cls.__instance = object.__new__(PVAContext)
+            cls.__instance.__initialized = False
+        return cls.__instance
+
+
+class Connection(PyDMConnection):
+
+    def __init__(self, channel, address, protocol=None, parent=None):
+        super(Connection, self).__init__(channel, address, protocol, parent)
+        conn = parse_channel_config(address, force_dict=True)
+        address = conn.get('parameters', {}).get('address')
```

Comment:
`conn['parameters']['address']` should be valid or it should blow up - just as in the pyepics one?